### PR TITLE
preparation protocol updates from Jean (SCP-2809)

### PR DESCRIPTION
**Pull request by devonbush**
Note that this does not handle any renaming of variables from construction->preparation.  I didn't want to step on Jon's toes since I think he's in this code too.  See https://docs.google.com/document/d/1tyXF_IMrgnmEgQZRph_5ryIoTFBNB9Z6YEyoQKbQz9k/edit#

### CODE REVIEW

File: app/models/expression_file_info.rb [hunk 1]
```diff
@@ -14,29 +14,43 @@ class ExpressionFileInfo
...
                                  '10x 5\' v2',
                                  '10x 5\' v3',
                                  'CEL-seq2',
                                  'Drop-seq',
-                                 'SCRB-seq',
-                                 'ATAC-seq',
-                                 'ChIP-seq',
-                                 'methylomics']
+                                 'inDrop',
+                                 'Mars-seq',
```
Comment (eweitz) [inline app/models/expression_file_info.rb:32]: Per similar comment in earlier PR, "MARS" is how it's rendered in literature:

```suggestion
                                 'MARS-seq',
```

**Review by eweitz — CHANGES_REQUESTED** [review]
Looks good, but for one user-facing nit.

File: app/models/expression_file_info.rb (hunk 1 above)
Comment (jlchang) [reply app/models/expression_file_info.rb:32]: Eric is correct - many thanks for catching that!

File: app/models/expression_file_info.rb [hunk 2]
```diff
@@ -14,29 +14,43 @@ class ExpressionFileInfo
...
+                                 'sci-RNA-seq',
+                                 'Seq-Well S^3',
+                                 'Seq-Well v1',
+                                 'Smart-like',
+                                 'Smart-seq2/Fluidigm C1',
+                                 # non-scRNAseq Assays
+                                 # scATACseq
```
Comment (jlchang) [inline app/models/expression_file_info.rb:39]:
```suggestion
                                 # single cell ATAC-seq assays
```

File: app/models/expression_file_info.rb [hunk 3]
```diff
@@ -14,29 +14,43 @@ class ExpressionFileInfo
...
+                                 'dsc-ATAC-seq',
+                                 'dsci-ATAC-seq',
+                                 'scATAC-seq/10x',
+                                 'scATAC-seq/Fluidigm',
+                                 'sci-ATAC-seq',
+                                 'scTHS-seq',
+                                 'snATAC-seq',
+                                 # spatial transcriptomics
```
Comment (jlchang) [inline app/models/expression_file_info.rb:47]:
```suggestion
                                 # spatial transcriptomics assays
```

File: app/models/expression_file_info.rb [hunk 4]
```diff
@@ -14,29 +14,43 @@ class ExpressionFileInfo
...
+                                 '10x Visium',
+                                 'MERFISH',
+                                 'osmFISH',
+                                 'SeqFISH+',
+                                 'Slide-seq',
+                                 'smFISH',
+                                 'Drop-ChIP']
```
Comment (jlchang) [inline app/models/expression_file_info.rb:54]: For consistency with comments above to indicate assay types would add, as a comment (above Drop-ChIP):
single cell ChIP-seq assays

File: app/models/expression_file_info.rb [hunk 5]
```diff
@@ -14,29 +14,43 @@ class ExpressionFileInfo
...
   validates :units, inclusion: {in: UNITS_VALUES}
 
-  BIOSAMPLE_INPUT_TYPE_VALUES = [nil, 'whole cell', 'single nuclei', 'bulk']
+  BIOSAMPLE_INPUT_TYPE_VALUES = [nil, 'Whole cell', 'Single nuclei', 'Bulk']
   validates :biosample_input_type, inclusion: {in: BIOSAMPLE_INPUT_TYPE_VALUES}
 
   MULTIMODALITY_VALUES = [nil, 'CITE-seq', 'Patch-seq']
   validates :multimodality, inclusion: {in: MULTIMODALITY_VALUES}
 
   LIBRARY_CONSTRUCTION_VALUES = [nil,
```
Comment (jlchang) [inline app/models/expression_file_info.rb:23]: We ultimately want `LIBRARY_CONSTRUCTION_VALUES` to be the same as library_preparation_protocol from the metadata convention (ie. if we accept values in both places, we would be checking the data matches for data integrity) not sure if we want/need these names to be the same in code right now. For the user facing text in the expression matrix upload wizard, we should definitely use "Library preparation protocol".

File: app/models/expression_file_info.rb [hunk 6]
```diff
@@ -14,29 +14,43 @@ class ExpressionFileInfo
...
-                                 'ATAC-seq',
-                                 'ChIP-seq',
-                                 'methylomics']
+                                 'inDrop',
+                                 'Mars-seq',
+                                 'sci-RNA-seq',
+                                 'Seq-Well S^3',
+                                 'Seq-Well v1',
+                                 'Smart-like',
+                                 'Smart-seq2/Fluidigm C1',
```
Comment (jlchang) [inline app/models/expression_file_info.rb:37]: 'Smart-seq2/plate-based' is missing, please add

**Review by jlchang — CHANGES_REQUESTED** [review]
Mostly minor suggestions, but the list is missing "Smart-seq2/plate-based" which must be added.

I note that `LIBRARY_CONSTRUCTION_VALUES` collected with expression matrices may ultimately be values we report for metadata convention `library_preparation_protocol`. Should the name be changed now (user-facing we should definitely use `Library preparation protocol`), is there value in being able to see "at-a-glance" whether the value was collected via expression matrix ingest or through the metadata file?

File: app/models/expression_file_info.rb [hunk 7]
```diff
@@ -14,29 +14,43 @@ class ExpressionFileInfo
   UNITS_VALUES = [nil, 'UMI-corrected raw counts', 'raw counts']
   validates :units, inclusion: {in: UNITS_VALUES}
 
-  BIOSAMPLE_INPUT_TYPE_VALUES = [nil, 'whole cell', 'single nuclei', 'bulk']
+  BIOSAMPLE_INPUT_TYPE_VALUES = [nil, 'Whole cell', 'Single nuclei', 'Bulk']
```
Comment (jlchang) [inline app/models/expression_file_info.rb:17]: These changes require the corresponding values for the synthetic studies also need updating:
db/seed/synthetic_studies/human_raw_counts/study_info.json#L21
db/seed/synthetic_studies/chicken_raw_counts/study_info.json#L21

File: app/models/expression_file_info.rb (hunk 5 above)
Comment (bistline) [reply app/models/expression_file_info.rb:23]: > We ultimately want `LIBRARY_CONSTRUCTION_VALUES` to be the same as library_preparation_protocol from the metadata convention (ie. if we accept values in both places, we would be checking the data matches for data integrity) not sure if we want/need these names to be the same in code right now. For the user facing text in the expression matrix upload wizard, we should definitely use "Library preparation protocol".

I've made this change on my PR: #776

**Review by bistline — APPROVED** [review]

**Review by eweitz — APPROVED** [review]

**Review by jlchang — APPROVED** [review]
Thanks for making the updates!

**Review by knapii-developments — APPROVED** [review]

**Pull request merged**